### PR TITLE
Enable light dismiss overlay for popups 2

### PR DESCRIPTION
**Pull request by maxkatz6**
Second attempt to fix bug with light dismiss overlay inside of popups. See https://github.com/AvaloniaUI/Avalonia/pull/4898:
> Fixed problem: a ComboBox inside of a Popup doesn't get dismissed if parent Popup got clicked on

Previous fix caused problems with Menu and ContextMenu, so it was reverted: https://github.com/AvaloniaUI/Avalonia/pull/4921.
The problem is with DefaultMenuInteractionHandler that listens for PointerLeave events on MenuItem, and when:
1. MenuItem clicked by user (or hovered).
2. Sub menu is opened.
3. It creates light dismiss layer.
4. Light dismiss layer steals pointer focus.
5. MenuItem from step 1 receives PointerLeave event.
6. DefaultMenuInteractionHandler closes sub menu.

With this PR MenuItem will not lose pointer focus because it's defined as submenu's OverlayInputPassThroughElement.
Fix is pretty clean, and seems to work without problems, but on other hand it feels like DefaultMenuInteractionHandler can have smarter checks instead.

### CODE REVIEW

File: src/Avalonia.Themes.Fluent/Controls/MenuItem.xaml [hunk 1]
```diff
@@ -112,6 +112,7 @@
           <Popup Name="PART_Popup"
                  WindowManagerAddShadowHint="False"
                  PlacementMode="Right"
+                 OverlayInputPassThroughElement="{Binding $parent[MenuItem]}"
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/MenuItem.xaml:115]: Probably it isn't required anymore after https://github.com/AvaloniaUI/Avalonia/pull/5322 was merged.
I need to test it later.

Comment (maxkatz6) [reply src/Avalonia.Themes.Fluent/Controls/MenuItem.xaml:115]: Checked, it's still required.

**Review by grokys — APPROVED** [review]
Yeah not sure if this should be handled in the interaction handler or not, but I'm leaning towards this making more sense.

**Comment by grokys** [comment]
This seems to have caused another problem: https://github.com/AvaloniaUI/Avalonia/issues/5475

Taking a look into potential fixes that don't involve disabling the light dismiss overlay layer for popups.

**Pull request merged**